bson/struct_codec.go

```diff
@@ -181,7 +181,9 @@ func (sc *StructCodec) EncodeValue(ec EncodeContext, vw ValueWriter, val reflect
 		encoder := desc.encoder
 
 		var empty bool
-		if cz, ok := encoder.(CodecZeroer); ok {
+		if cz, ok := encoder.(interface {
+			IsTypeZero(interface{}) bool
+		}); ok {
 			empty = cz.IsTypeZero(rv.Interface())
```

Comment:
We should consider completely removing the `IsTypeZero` feature. Based on [searching for uses in open-source repos](https://sourcegraph.com/search?q=context:global+lang:Go++-repo:%5Egithub%5C.com/mongodb/mongo-go-driver%24+-file:%28%5E%7C/%29vendor/+IsTypeZero&patternType=standard&case=yes&sm=0&groupBy=repo), it seems completely unused.